src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -33,17 +33,15 @@ public class ModuleIntrinsics
         [TraceSource("Modules", "Module loading and analysis")]
```

Comment:
@iSazonov, your last commit had 2 failures in `PowerShell-CI-macos`
Get-Module -ListAvailable.Get-Module -FullyQualifiedName @{ModuleName = 'Microsoft.PowerShell.Managemen*' ; ModuleVersion = '7.0.0.0'} - Wildcard
```powershell
Expected a collection with size 1, but got an empty collection.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Get-Module.Tests.ps1: line 183
183:         $modules | Should -HaveCount 1
```
Get-Module -ListAvailable.Get-Module -FullyQualifiedName @{ModuleName = 'microsoft.powershell.managemen*' ; ModuleVersion = '7.0.0.0'} - Mismatched case
```powershell
Expected a collection with size 1, but got an empty collection.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Get-Module.Tests.ps1: line 183
183:         $modules | Should -HaveCount 1
```